src/server/svr_comm.c

```diff
@@ -310,6 +310,112 @@ send_nodestat_req(enum msvr_stats_type type)
 	mtfd = -1;
 }
 
+/**
+ * @brief broadcast discard job request to peer servers
+ * 
+ * @param[in] jobid - job id
+ * @param execvnode - job's exec vnode
+ * @param exechost - job's exec host
+ * @param rver - job run version
+ * 
+ * @return int 
+ * @retval 0: success
+ * @retval !0: failure
+ */
+int
+ps_send_discard(char *jobid, char *execvnode, char *exechost, int rver)
+{
+	int mtfd = -1;
+	int rc;
+
+	if (!exechost)
+		return -1;
+
+	mtfd = open_ps_mtfd();
```

Comment:
so which peer server will this request be sent to?

Reply:
The server will broadcast the request to all peer servers. We should be able to optimize it by reusing an already opened stream by saving it in the job struct.

Reply:
The job might span over multiple servers so we'd need to store multiple streams .. let's wait until we see a performance problem and then fix this, for now broadcast should be ok.

Reply:
It will be a multicast fd. no need to store multiple streams. But we could take it up separately.